modules/dnn/perf/perf_net.cpp

```diff
@@ -29,10 +29,7 @@ class DNNTestNetwork : public ::perf::TestBaseWithParam< tuple<Backend, Target>
     }
 
     void processNet(std::string weights, std::string proto, std::string halide_scheduler,
-                    const Mat& input, const std::string& outputLayer = "")
-    {
-        randu(input, 0.0f, 1.0f);
-
+                    const std::vector<std::tuple<Mat, std::string>>& inputs, const std::string& outputLayer = ""){
```

Comment:
Why not `std::map`?

Reply:
Because `net.getMemoryConsumption()` need the input shape in order. Map will destroy the sequence.